github_server/action.yml

```
@@ -1,53 +1,143 @@
-    required: false
-    # no default to let the custom Scenario timeouts create the reload timeout
+    default: 600  # 10 min
+
+outputs:
+  DOCASSEMBLE_DEVELOPER_API_KEY:
+    description: "Use this inside `with:` for the ALKiln step. Temporary admin api key. It will be destroyed after the tests are done."
+    value: ${{ steps.api_key_output_step.outputs.DOCASSEMBLE_DEVELOPER_API_KEY }}
+  SERVER_URL:
+    description: "Use this inside `with:` for the ALKiln step. URL of the temporary docassemble server."
+    value: "http://localhost"
+  DA_ADMIN_EMAIL:
+    value: ${{ steps.api_key_output_step.outputs.DA_ADMIN_EMAIL }}
+  DA_ADMIN_PASSWORD:
+    value: ${{ steps.api_key_output_step.outputs.DA_ADMIN_PASSWORD }}
 
 
 runs:
   using: "composite"
   steps:
-      - name: "github_server: Set environment variables"
+      - uses: actions/checkout@v3
+
+      - name: "ALKiln github_server: Set environment variables"
         run: |
           echo "DA_ADMIN_API_KEY=abcd1234abcd1234abcd5678abdc5678" >> $GITHUB_ENV
+          echo "MAX_SECONDS_FOR_DOCKER=${{ inputs.MAX_SECONDS_FOR_DOCKER }}" >> $GITHUB_ENV
+          echo "DA_ADMIN_EMAIL=admin@example.com" >> $GITHUB_ENV
+          # Login will fail if this is "password"
+          echo "DA_ADMIN_PASSWORD=@123abcdefg" >> $GITHUB_ENV
         shell: bash
-      - name: Download and start docker
+      - id: api_key_output_step
+        run: |
+          echo "DOCASSEMBLE_DEVELOPER_API_KEY=$DA_ADMIN_API_KEY" >> "$GITHUB_OUTPUT"
+          echo "DA_ADMIN_EMAIL=$DA_ADMIN_EMAIL" >> "$GITHUB_OUTPUT"
+          echo "DA_ADMIN_PASSWORD=$DA_ADMIN_PASSWORD" >> "$GITHUB_OUTPUT"
+        shell: bash
+
+      - name: "ALKiln github_server: Download and start docker"
         run: |
           mkdir /tmp/config
-          echo "$CONFIG_CONTENTS" > /tmp/config/myconfig.yml 
-          docker pull brycewilley/smoldocassemble
-          docker run --env DA_ADMIN_EMAIL="admin@example.com" --env DA_ADMIN_PASSWORD="password" --env DA_ADMIN_API_KEY="${{ env.DA_ADMIN_API_KEY }}" --env DA_CONFIG=/tmp/config/myconfig.yml --volume /tmp/config:/tmp/config --cap-add SYS_PTRACE --memory="4gb" -d -p 80:80 jhpyle/docassemble
-          # TODO before merge: do a while loop here until `localhost` shows "It's ready"
-          sleep 480
-          curl localhost
-        shell: bash
-      - name: Run Kiln
-        # TODO: Research how to get this to always be the same branch we're on
-        uses: SuffolkLITLab/ALKiln@v5
+          echo "${{ inputs.CONFIG_CONTENTS }}" > /tmp/config/myconfig.yml
+          docker pull jhpyle/docassemble
+          docker run --env DA_ADMIN_EMAIL="${{ env.DA_ADMIN_EMAIL }}" --env DA_ADMIN_PASSWORD="${{ env.DA_ADMIN_PASSWORD }}" --env DA_ADMIN_API_KEY="${{ env.DA_ADMIN_API_KEY }}" --env DA_CONFIG=/tmp/config/myconfig.yml --volume /tmp/config:/tmp/config --cap-add SYS_PTRACE --memory="4gb" -d -p 80:80 jhpyle/docassemble
```

```suggestion
          docker run --env DA_ADMIN_EMAIL="${{ env.DA_ADMIN_EMAIL }}" --env DA_ADMIN_PASSWORD="${{ env.DA_ADMIN_PASSWORD }}" --env DA_ADMIN_API_KEY="${{ env.DA_ADMIN_API_KEY }}" --env DA_CONFIG=/tmp/config/myconfig.yml --volume /tmp/config:/tmp/config --cap-add SYS_PTRACE --memory="4gb" -d -p 127.0.0.1:80:80 jhpyle/docassemble
```

This will make sure that the docassemble server is only visible from the GitHub action machine, a bit more secure.

---

Note: We set `--memory` to give a more understandable error if docker runs out of memory, but we may in future need to add an input for that value if someone has a problem.

---

If we give it `--name docassemble_container`, would that be a useful output so devs don't go have to go looking for the process name? Wish we could get it's id too.

---

I think we should suggest people use `docker ps` command

---

Note: made #801 to follow up